inlang/source-code/paraglide/paraglide-js/src/compilerV2/compile.ts

```diff
@@ -0,0 +1,143 @@
+ * a map of file names to file contents.
+ *
+ * @example
+ *   const output = compile({ messages, settings })
+ *   console.log(output)
+ *   >> { "messages.js": "...", "runtime.js": "..." }
+ */
+export const compile = async (args: CompileOptions): Promise<Record<string, string>> => {
+	const opts = {
+		...defaultCompileOptions,
+		...args,
+	}
+
+	//Maps each language to it's fallback
+	//If there is no fallback, it will be undefined
+	const fallbackMap = getFallbackMap(opts.settings.locales, opts.settings.baseLocale)
+
+	const resources = opts.bundles.map((bundle) => compileBundle(bundle, fallbackMap))
+
+	const indexFile = [
+		"/* eslint-disable */",
+		'import { languageTag } from "./runtime.js"',
+		opts.settings.locales
+			.map((locale) => `import * as ${i(locale)} from "./messages/${locale}.js"`)
+			.join("\n"),
+		"\n",
+		resources.map(({ bundle }) => bundle.code).join("\n\n"),
+	].join("\n")
+
+	const output: Record<string, string> = {
+		".prettierignore": ignoreDirectory,
+		".gitignore": ignoreDirectory,
+		"runtime.js": await fmt(createRuntime(opts.settings)),
+		"registry.js": await fmt(createRegistry()),
+		"messages.js": await fmt(indexFile),
+	}
+
+	// generate message files
+	for (const locale of opts.settings.locales) {
+		const filename = `messages/${locale}.js`
+		let file = ["/* eslint-disable */", "import * as registry from '../registry.js' "].join("\n")
+
+		for (const resource of resources) {
+			// Bundle Aliases
+			const alias = resource.bundle.source.alias
+				? Object.values(resource.bundle.source.alias).at(0)
+				: undefined
+
+			const compiledMessage = resource.messages[locale]
+			if (!compiledMessage) {
+				// add fallback
+				const fallbackLocale = fallbackMap[locale]
+				if (fallbackLocale) {
+					file += `\nexport { ${i(resource.bundle.source.id)} } from "./${fallbackLocale}.js"`
+					if (alias) file += `\nexport { ${i(alias)} } from "./${fallbackLocale}.js"`
+				} else {
+					file += `\nexport const ${i(resource.bundle.source.id)} = '${escapeForSingleQuoteString(
+						resource.bundle.source.id
+					)}'`
+					if (alias) file += `\nexport {${i(resource.bundle.source.id)} as ${i(alias)}}`
+				}
+
+				continue
+			}
+
+			file += `\n${compiledMessage.code}`
+			file += `\nexport { ${i(compiledMessage.source.id)} as ${resource.bundle.source.id} }`
+			if (alias) file += `\nexport { ${i(compiledMessage.source.id)} as ${i(alias)} }`
+		}
+
+		output[filename] = await fmt(file)
+	}
+
+	// telemetry
+	const pkgJson = await getPackageJson(fs, process.cwd())
+	const stack = getStackInfo(pkgJson)
+	telemetry.capture(
+		{
+			event: "PARAGLIDE-JS compile executed",
+			properties: { stack },
+		},
+		opts.projectId
+	)
+
+	telemetry.shutdown()
+	return output
+}
+
+async function fmt(js: string): Promise<string> {
+	return await prettier.format(js, {
+		arrowParens: "always",
+		singleQuote: true,
+		parser: "babel",
+		plugins: ["prettier-plugin-jsdoc"],
+	})
+}
```

Comment:
(No action required)

Such an easy workaround to get formatting right :+1. 

Besides, I find reading source code easier than an AST framework. I hope that we get around an AST. It makes contributions, reviews, etc. easier.